gpiozero/mixins.py

```diff
@@ -270,13 +280,119 @@ def _fire_events(self):
             if not old_state and new_state:
```

Comment:
Would this be neater if the `else:` on the line above this was changed to `elif old_state != new_state:`, and then `if not old_state and new_state:` could be changed to just `if new_state:` and `elif old_state and not new_state:` could be changed to `else:`
?